pkg/cloud/openstack/machine/actuator.go

```diff
@@ -149,7 +155,13 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 	}
 
 	if machineProviderSpec.FloatingIP != "" {
-		err := computeService.AssociateFloatingIP(instance.ID, machineProviderSpec.FloatingIP)
+		err := networkingService.CreateFloatingIP(clusterProviderSpec, machineProviderSpec.FloatingIP)
```

Comment:
my env has some issue here
rule:create_floatingip and rule:create_floatingip:floating_ip_address) is disallowed by policy"
so I can't test, but code looks fine

Reply:
But it works if the floating ip is already there before?

Reply:
yes, you are right, I tested that path ,so I think we are ready to go now

Reply:
Perfect. Didn't want to break your use case with that :)